isis/cmake/BuildDocs.cmake

```diff
@@ -66,22 +66,30 @@ endfunction(copy_app_docs_info)
 # Build the top level of the documents directory
 function(build_upper_level)
 
-  # Copy existing folders to the install directory
-  copy_folder(${docBuildFolder}/assets ${docInstallFolder})
-  copy_folder(${docBuildFolder}/w3c    ${docInstallFolder})
-
   # Make new (empty) output folders
   set(newFolders UserDocs AboutIsis General Guides Installation TechnicalInfo)
   foreach(f ${newFolders})
-    file(MAKE_DIRECTORY "${docInstallFolder}/${f}")
+    file(MAKE_DIRECTORY "${docInstallFolder}/${docVersion}/${f}")
   endforeach()
 
+  # Copy the assets folder to the specific version directory
+  copy_folder(${docBuildFolder}/assets ${docInstallFolder}/${docVersion})
+  # Also copy the assest folder to the base area so the base index.html can use it
+  # TODO - eliminate this copy of assest by having a simple top index.html that redirects to the vx.y.z/index.html. The top index should contain html base so it uses vx.y.z assets
+  copy_folder(${docBuildFolder}/assets ${docInstallFolder})
+
+  # Copy other folders to the specific version directory
+  copy_folder(${docBuildFolder}/w3c    ${docInstallFolder}/${docVersion})
+
   # These folders are populated inside "build_documents_folder"
 
-  # Create index.html file
-  execute_process(COMMAND ${XALAN} ${XALAN_VALIDATE_OPTION} ${XALAN_PARAM_OPTION} menuPath \"\" ${XALAN_OUTFILE_OPTION} ${docInstallFolder}/index.html ${XALAN_INFILE_OPTION} ${docBuildFolder}/build/homepage.xml ${XALAN_XSL_OPTION} ${docBuildFolder}/build/main.xsl)
+  # Create the main index.html file aka the home page
+  execute_process(COMMAND ${XALAN} ${XALAN_VALIDATE_OPTION} ${XALAN_PARAM_OPTION} menuPath \"${docVersion}/\" ${XALAN_OUTFILE_OPTION} ${docInstallFolder}/index.html ${XALAN_INFILE_OPTION} ${docBuildFolder}/build/homepage.xml ${XALAN_XSL_OPTION} ${docBuildFolder}/build/main.xsl)
+  # Create a second main page inside the version numbered area for when the page above gets overwritten with a new version 
+  execute_process(COMMAND ${XALAN} ${XALAN_VALIDATE_OPTION} ${XALAN_PARAM_OPTION} menuPath \"\" ${XALAN_OUTFILE_OPTION} ${docInstallFolder}/${docVersion}/index.html ${XALAN_INFILE_OPTION} ${docBuildFolder}/build/homepage.xml ${XALAN_XSL_OPTION} ${docBuildFolder}/build/main.xsl)
```

Comment:
So here, the plan is to make two identical doc builds that will get pushed up to the server. The first being the main homepage you navigate to when you go to isis.astrogeology.usgs.gov and the other being the homepage you navigate to if you click on the current version in the drop down menu?

Reply:
Just two identical index.html files. One in the rood of isis.astro.usgs.gov and the other in v7.0.0. This way 7.0.0 never goes away and can be fully accessed, but the root index file gets renewed every time we push a new version.  I have ideas on how we can fix this, but haven't been able to test yet.